sdk/storage/azure-storage-blobs/src/blob_client.cpp

```diff
@@ -46,15 +46,14 @@ namespace Azure { namespace Storage { namespace Blobs {
       : BlobClient(blobUrl, options)
   {
     Azure::Core::Http::TelemetryPolicyOptions telemetryPolicyOptions;
-    telemetryPolicyOptions.ApplicationId = options.ApplicationId;
+    telemetryPolicyOptions.ApplicationId = options.Telemetry.ApplicationId;
     m_pipeline = std::make_shared<Azure::Core::Internal::Http::HttpPipeline>(
-        Storage::Details::ConstructPolicies(
-            std::make_unique<Azure::Core::Http::TelemetryPolicy>(
-                Storage::Details::BlobServicePackageName,
-                Details::Version::VersionString(),
-                telemetryPolicyOptions),
-            std::make_unique<Storage::Details::SharedKeyPolicy>(credential),
-            options));
+        options,
+        Storage::Details::BlobServicePackageName,
+        Details::Version::VersionString(),
+        Storage::Details::ConstructPerOperationPolicies(nullptr, options),
+        Storage::Details::ConstructPerRetryPolicies(
+            std::make_unique<Storage::Details::SharedKeyPolicy>(credential)));
```

Comment:
The order of storage policies is delicate sometimes very tricky. I appreciate your help in making things work with the new client options. But I think it'd be better to undo these changes and let storage developer take care of it.

Reply:
Yeah, I can undo.
I wanted to provide some hints on how to call the new HttpPipeline constructor. But yes, please do not consider it as the final and desired fix to storage.

Reply:
Your `HttpPipepipeline` constructor won't work for storage scenario. Auth policy has to be the last one that can modify HTTP url or request headers.

Reply:
I have updated the order to let service's policies to execute after custom-end-user policies.
That should be enough. (The logging policy at the end won't update the request. It only read from it).